Client tls policy

maphad opened the pull request:
Add configuration for traffic director clientTlsPolicy resource

part of https://github.com/hashicorp/terraform-provider-google/issues/14040


<!--
Replace each [ ] with [X] to check it. Switch to the preview view to make it easier to click on links.
These steps will speed up the review process, and we appreciate you spending time on them before sending
your code to be reviewed.
-->
If this PR is for Terraform, I acknowledge that I have:

- [ x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [ x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

<!-- AUTOCHANGELOG for Downstream PRs.

Please select one of the following "release-note:" headings:
    - release-note:enhancement
    - release-note:bug
    - release-note:note
    - release-note:new-resource
    - release-note:new-datasource
    - release-note:deprecation
    - release-note:breaking-change
    - release-note:none

Unless you choose release-note:none, please add a release note.

See .ci/RELEASE_NOTES_GUIDE.md for writing good release notes.

You can add more release note blocks if you want more than one CHANGELOG
entry for this PR.
-->
**Release Note Template for Downstream PRs (will be copied)**


```release-note:new-resource
`google_networksecurity_client_tls_policy`
```

modular-magician commented:
Hello!  I am a robot who works on Magic Modules PRs.

I've detected that you're a community contributor. @shuyama1, a repository maintainer, has been assigned to assist you and help review your changes. 

<details>
  <summary>:question: First time contributing? Click here for more details</summary>

---

Your assigned reviewer will help review your code by: 
* Ensuring it's backwards compatible, covers common error cases, etc.
* Summarizing the change into a user-facing changelog note.
* Passes tests, either our "VCR" suite, a set of presubmit tests, or with manual test runs.

You can help make sure that review is quick by running local tests and ensuring they're passing in between each push you make to your PR's branch. Also, try to leave a comment with each push you make, as pushes generally don't generate emails.

If your reviewer doesn't get back to you within a week after your most recent change, please feel free to leave a comment on the issue asking them to take a look! In the absence of a dedicated review dashboard most maintainers manage their pending reviews through email, and those will sometimes get lost in their inbox.

---

</details>

shuyama1 commented:
Reassigning this PR as it's still WIP and I will be out for the next week.
/gcbrun

modular-magician commented:
Hello!  I am a robot who works on Magic Modules PRs.

I've detected that you're a community contributor. @zli82016, a repository maintainer, has been assigned to assist you and help review your changes. 

<details>
  <summary>:question: First time contributing? Click here for more details</summary>

---

Your assigned reviewer will help review your code by: 
* Ensuring it's backwards compatible, covers common error cases, etc.
* Summarizing the change into a user-facing changelog note.
* Passes tests, either our "VCR" suite, a set of presubmit tests, or with manual test runs.

You can help make sure that review is quick by running local tests and ensuring they're passing in between each push you make to your PR's branch. Also, try to leave a comment with each push you make, as pushes generally don't generate emails.

If your reviewer doesn't get back to you within a week after your most recent change, please feel free to leave a comment on the issue asking them to take a look! In the absence of a dedicated review dashboard most maintainers manage their pending reviews through email, and those will sometimes get lost in their inbox.

---

</details>

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-7807-old..auto-pr-7807) ( 1 file changed, 181 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-7807-old..auto-pr-7807) ( 5 files changed, 1180 insertions(+), 2 deletions(-))
TF Conversion: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-7807-old..auto-pr-7807) ( 2 files changed, 3 insertions(+), 3 deletions(-))
TF OiCS: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-7807-old..auto-pr-7807) ( 4 files changed, 110 insertions(+))

modular-magician commented:
#### Tests analytics 
Total tests: `2671` 
Passed tests `2376` 
Skipped tests: `280` 
Affected tests: `15` 

#### Action taken 
<details> <summary>Found 15 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccFirebaserulesRelease_BasicRelease|TestAccComputeRouterBgpPeer_routerPeerRouterApplianceExample|TestAccComputeRouterPeer_enable|TestAccComputeRouterPeer_advertiseMode|TestAccComputeRouterPeer_basic|TestAccComputeRouterPeer_routerApplianceInstance|TestAccComputeRouterPeer_bfd|TestAccAlloydbCluster_missingLocation|TestAccApigeeKeystoresAliasesKeyCertFile_apigeeKeystoresAliasesKeyCertFileTestExample|TestAccAlloydbBackup_missingLocation|TestAccApigeeKeystoresAliasesPkcs12_ApigeeKeystoresAliasesPkcs12Example|TestAccNetworkSecurityClientTlsPolicy_networkSecurityClientTlsPolicyBasicExample|TestAccDataSourceGoogleFirebaseAndroidAppConfig|TestAccDataSourceComputeRouterStatus|TestAccDataSourceAlloydbLocations_basic </blockquote></details> 

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
Tests passed during RECORDING mode:
 `TestAccFirebaserulesRelease_BasicRelease`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/197ab092-8808-4968-af4a-a4350eceda2c/recording/TestAccFirebaserulesRelease_BasicRelease.log)]
`TestAccComputeRouterBgpPeer_routerPeerRouterApplianceExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/197ab092-8808-4968-af4a-a4350eceda2c/recording/TestAccComputeRouterBgpPeer_routerPeerRouterApplianceExample.log)]
`TestAccComputeRouterPeer_enable`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/197ab092-8808-4968-af4a-a4350eceda2c/recording/TestAccComputeRouterPeer_enable.log)]
`TestAccComputeRouterPeer_advertiseMode`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/197ab092-8808-4968-af4a-a4350eceda2c/recording/TestAccComputeRouterPeer_advertiseMode.log)]
`TestAccComputeRouterPeer_routerApplianceInstance`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/197ab092-8808-4968-af4a-a4350eceda2c/recording/TestAccComputeRouterPeer_routerApplianceInstance.log)]
`TestAccComputeRouterPeer_bfd`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/197ab092-8808-4968-af4a-a4350eceda2c/recording/TestAccComputeRouterPeer_bfd.log)]
`TestAccAlloydbCluster_missingLocation`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/197ab092-8808-4968-af4a-a4350eceda2c/recording/TestAccAlloydbCluster_missingLocation.log)]
`TestAccApigeeKeystoresAliasesKeyCertFile_apigeeKeystoresAliasesKeyCertFileTestExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/197ab092-8808-4968-af4a-a4350eceda2c/recording/TestAccApigeeKeystoresAliasesKeyCertFile_apigeeKeystoresAliasesKeyCertFileTestExample.log)]
`TestAccAlloydbBackup_missingLocation`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/197ab092-8808-4968-af4a-a4350eceda2c/recording/TestAccAlloydbBackup_missingLocation.log)]
`TestAccApigeeKeystoresAliasesPkcs12_ApigeeKeystoresAliasesPkcs12Example`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/197ab092-8808-4968-af4a-a4350eceda2c/recording/TestAccApigeeKeystoresAliasesPkcs12_ApigeeKeystoresAliasesPkcs12Example.log)]
`TestAccNetworkSecurityClientTlsPolicy_networkSecurityClientTlsPolicyBasicExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/197ab092-8808-4968-af4a-a4350eceda2c/recording/TestAccNetworkSecurityClientTlsPolicy_networkSecurityClientTlsPolicyBasicExample.log)]
`TestAccDataSourceComputeRouterStatus`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/197ab092-8808-4968-af4a-a4350eceda2c/recording/TestAccDataSourceComputeRouterStatus.log)]
`TestAccDataSourceAlloydbLocations_basic`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/197ab092-8808-4968-af4a-a4350eceda2c/recording/TestAccDataSourceAlloydbLocations_basic.log)] 

Tests failed during RECORDING mode:
 `TestAccComputeRouterPeer_basic`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/197ab092-8808-4968-af4a-a4350eceda2c/build-log/recording_build/TestAccComputeRouterPeer_basic_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/197ab092-8808-4968-af4a-a4350eceda2c/recording/TestAccComputeRouterPeer_basic.log)]
`TestAccDataSourceGoogleFirebaseAndroidAppConfig`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/197ab092-8808-4968-af4a-a4350eceda2c/build-log/recording_build/TestAccDataSourceGoogleFirebaseAndroidAppConfig_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/197ab092-8808-4968-af4a-a4350eceda2c/recording/TestAccDataSourceGoogleFirebaseAndroidAppConfig.log)] 

Please fix these to complete your PR
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/197ab092-8808-4968-af4a-a4350eceda2c/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/197ab092-8808-4968-af4a-a4350eceda2c/recording) for each test

modular-magician commented:
Tests passed during RECORDING mode:
 `TestAccDataSourceGoogleFirebaseAndroidAppConfig`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/13787c93-ab50-4409-8310-0d958f4be6cb/recording/TestAccDataSourceGoogleFirebaseAndroidAppConfig.log)]
`TestAccNetworkSecurityClientTlsPolicy_update`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/13787c93-ab50-4409-8310-0d958f4be6cb/recording/TestAccNetworkSecurityClientTlsPolicy_update.log)]
`TestAccNetworkSecurityClientTlsPolicy_networkSecurityClientTlsPolicyAdvancedExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/13787c93-ab50-4409-8310-0d958f4be6cb/recording/TestAccNetworkSecurityClientTlsPolicy_networkSecurityClientTlsPolicyAdvancedExample.log)]
`TestAccDataSourceAlloydbLocations_basic`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/13787c93-ab50-4409-8310-0d958f4be6cb/recording/TestAccDataSourceAlloydbLocations_basic.log)]
`TestAccAlloydbBackup_missingLocation`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/13787c93-ab50-4409-8310-0d958f4be6cb/recording/TestAccAlloydbBackup_missingLocation.log)]
`TestAccAlloydbCluster_missingLocation`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/13787c93-ab50-4409-8310-0d958f4be6cb/recording/TestAccAlloydbCluster_missingLocation.log)]
`TestAccApigeeKeystoresAliasesPkcs12_ApigeeKeystoresAliasesPkcs12Example`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/13787c93-ab50-4409-8310-0d958f4be6cb/recording/TestAccApigeeKeystoresAliasesPkcs12_ApigeeKeystoresAliasesPkcs12Example.log)]
`TestAccApigeeKeystoresAliasesKeyCertFile_apigeeKeystoresAliasesKeyCertFileTestExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/13787c93-ab50-4409-8310-0d958f4be6cb/recording/TestAccApigeeKeystoresAliasesKeyCertFile_apigeeKeystoresAliasesKeyCertFileTestExample.log)] 

All tests passed
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/13787c93-ab50-4409-8310-0d958f4be6cb/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/13787c93-ab50-4409-8310-0d958f4be6cb/recording) for each test

zli82016 commented on mmv1/products/networksecurity/ClientTlsPolicy.yaml:
```diff
@@ -0,0 +1,160 @@
+description: |
+  ClientTlsPolicy is a resource that specifies how a client should authenticate connections to backends of a service. This resource itself does not affect configuration unless it is attached to a backend service resource.
+references: !ruby/object:Api::Resource::ReferenceLinks
+  api: 'https://cloud.google.com/traffic-director/docs/reference/network-security/rest/v1beta1/projects.locations.clientTlsPolicies'
```
Is the user guild available to add to the resource?

maphad replied:
Added

zli82016 commented on mmv1/products/networksecurity/ClientTlsPolicy.yaml:
```diff
@@ -0,0 +1,160 @@
+# Copyright 2023 Google Inc.
```
This file should be put into the folder products/networkservices.

maphad replied:
Why is that? This is part of the networksecurity API.

zli82016 replied:
Oh. Yes, you are right. Sorry about that.

zli82016 commented on mmv1/third_party/terraform/tests/resource_network_security_client_tls_policy_test.go.erb:
```diff
@@ -0,0 +1,69 @@
+		CheckDestroy: testAccCheckNetworkSecurityClientTlsPolicyDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+        Config: testAccNetworkSecurityClientTlsPolicy_basic(clientTlsPolicyName),
```
Indentation is weird here.

maphad replied:
fixed

zli82016 commented on mmv1/templates/terraform/examples/network_security_client_tls_policy_advanced.tf.erb:
```diff
@@ -0,0 +1,19 @@
+        plugin_instance = "google_cloud_private_spiffe"
+      }
+    }
+  server_validation_ca {
```
As serverValidationCa is an array, can you please add more server_validation_ca blocks here? Thanks.

maphad replied:
done

zli82016 commented on mmv1/third_party/terraform/tests/resource_network_security_client_tls_policy_test.go.erb:
```diff
@@ -0,0 +1,69 @@
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccNetworkSecurityClientTlsPolicy_update(t *testing.T) {
```
Can you please add the updating tests for other fields, too? Thanks.

maphad replied:
Added

zli82016 commented on mmv1/templates/terraform/examples/network_security_client_tls_policy_basic.tf.erb:
```diff
@@ -0,0 +1,9 @@
+resource "google_network_security_client_tls_policy" "<%= ctx[:primary_resource_id] %>" {
```
The resource name should be google_network_services_security_client_tls_policy as it is a product in the service networkservices.

maphad replied:
The product is in network security

zli82016 replied:
Oh. Yes, you are right. Sorry about that.

zli82016 commented on mmv1/third_party/terraform/tests/resource_network_security_client_tls_policy_test.go.erb:
```diff
@@ -0,0 +1,94 @@
+    }
+    description            = "my description"
+    sni                    = "secure.example.com"
+		client_certificate {
+    certificate_provider_instance {
+        plugin_instance = "google_cloud_private_spiffe"
+      }
+    }
+  	server_validation_ca {
+    	grpc_endpoint {
+      	target_uri = "unix:mypath"
+    	}
+  	}
```
Can the indentation be aligned?

maphad replied:
Done!

zli82016 commented on mmv1/third_party/terraform/tests/resource_network_security_client_tls_policy_test.go.erb:
```diff
@@ -0,0 +1,94 @@
+    }
+    description            = "updated description"
+    sni                    = "secure1.example.com"
+		client_certificate {
+    certificate_provider_instance {
+        plugin_instance = "google_cloud"
+      }
+    }
+  	server_validation_ca {
+    	grpc_endpoint {
+      	target_uri = "unix:mypath1"
+    	}
+  	}
+		server_validation_ca {
+    	grpc_endpoint {
+      	target_uri = "unix:mypath2"
+    	}
+  	}
```
Can the indentation be aligned?

maphad replied:
Done!

zli82016 commented:
Thanks for making the changes. Mostly LGTM. Just added comments for the indentations.

maphad commented:
> Thanks for making the changes. Mostly LGTM. Just added comments for the indentations.

Thanks for review! fixed the indentation, something to do with my editor messing up when I copy. Thanks!

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-7807-old..auto-pr-7807) ( 2 files changed, 216 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-7807-old..auto-pr-7807) ( 6 files changed, 1359 insertions(+), 2 deletions(-))
TF Conversion: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-7807-old..auto-pr-7807) ( 2 files changed, 3 insertions(+), 3 deletions(-))
TF OiCS: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-7807-old..auto-pr-7807) ( 8 files changed, 234 insertions(+))

modular-magician commented:
#### Tests analytics 
Total tests: `2692` 
Passed tests `2404` 
Skipped tests: `279` 
Affected tests: `9` 

#### Action taken 
<details> <summary>Found 9 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccNetworkSecurityClientTlsPolicy_update|TestAccNetworkSecurityClientTlsPolicy_networkSecurityClientTlsPolicyAdvancedExample|TestAccFirebaserulesRelease_BasicRelease|TestAccDataSourceGoogleFirebaseAndroidAppConfig|TestAccDataSourceAlloydbLocations_basic|TestAccAlloydbBackup_missingLocation|TestAccAlloydbCluster_missingLocation|TestAccApigeeKeystoresAliasesPkcs12_ApigeeKeystoresAliasesPkcs12Example|TestAccApigeeKeystoresAliasesKeyCertFile_apigeeKeystoresAliasesKeyCertFileTestExample </blockquote></details> 

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
Tests passed during RECORDING mode:
 `TestAccNetworkSecurityClientTlsPolicy_update`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/3fc492b5-8266-47dc-a0a1-5222a3bc710b/recording/TestAccNetworkSecurityClientTlsPolicy_update.log)]
`TestAccNetworkSecurityClientTlsPolicy_networkSecurityClientTlsPolicyAdvancedExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/3fc492b5-8266-47dc-a0a1-5222a3bc710b/recording/TestAccNetworkSecurityClientTlsPolicy_networkSecurityClientTlsPolicyAdvancedExample.log)]
`TestAccFirebaserulesRelease_BasicRelease`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/3fc492b5-8266-47dc-a0a1-5222a3bc710b/recording/TestAccFirebaserulesRelease_BasicRelease.log)]
`TestAccDataSourceAlloydbLocations_basic`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/3fc492b5-8266-47dc-a0a1-5222a3bc710b/recording/TestAccDataSourceAlloydbLocations_basic.log)]
`TestAccAlloydbBackup_missingLocation`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/3fc492b5-8266-47dc-a0a1-5222a3bc710b/recording/TestAccAlloydbBackup_missingLocation.log)]
`TestAccAlloydbCluster_missingLocation`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/3fc492b5-8266-47dc-a0a1-5222a3bc710b/recording/TestAccAlloydbCluster_missingLocation.log)]
`TestAccApigeeKeystoresAliasesPkcs12_ApigeeKeystoresAliasesPkcs12Example`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/3fc492b5-8266-47dc-a0a1-5222a3bc710b/recording/TestAccApigeeKeystoresAliasesPkcs12_ApigeeKeystoresAliasesPkcs12Example.log)]
`TestAccApigeeKeystoresAliasesKeyCertFile_apigeeKeystoresAliasesKeyCertFileTestExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/3fc492b5-8266-47dc-a0a1-5222a3bc710b/recording/TestAccApigeeKeystoresAliasesKeyCertFile_apigeeKeystoresAliasesKeyCertFileTestExample.log)] 

Tests failed during RECORDING mode:
 `TestAccDataSourceGoogleFirebaseAndroidAppConfig`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/3fc492b5-8266-47dc-a0a1-5222a3bc710b/build-log/recording_build/TestAccDataSourceGoogleFirebaseAndroidAppConfig_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/3fc492b5-8266-47dc-a0a1-5222a3bc710b/recording/TestAccDataSourceGoogleFirebaseAndroidAppConfig.log)] 

Please fix these to complete your PR
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/3fc492b5-8266-47dc-a0a1-5222a3bc710b/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-7807/artifacts/3fc492b5-8266-47dc-a0a1-5222a3bc710b/recording) for each test

zli82016 reviewed (approved):
LGTM. Thanks.
